plugins/hypervisors/hyperv/src/com/cloud/hypervisor/hyperv/resource/HypervDirectConnectResource.java

```diff
@@ -594,6 +597,25 @@ private PlugNicAnswer execute(final PlugNicCommand cmd) {
         }
     }
 
+    private CopyFileInVmAnswer execute(CopyFileInVmCommand cmd) {
+        File keyFile = getSystemVMKeyFile();
+        try {
+            File file = new File(cmd.getSrc());
+            if(file.exists()) {
+                if(file.isDirectory()) {
+                    for (File f : FileUtils.listFiles(new File(cmd.getSrc()), TrueFileFilter.INSTANCE, TrueFileFilter.INSTANCE)) {
+                        SshHelper.scpTo(cmd.getVmIp(), 3922, "root", keyFile, null, cmd.getDest(), f.getCanonicalPath(), null);
+                    }
+                } else {
+                    SshHelper.scpTo(cmd.getVmIp(), 3922, "root", keyFile, null, cmd.getDest(), file.getCanonicalPath(), null);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
SshHelper is throwing Exception so I am catching Exception. If you are talking about handling it here and not passing to CopyFileInVmAnswer then CopyFileInVmAnswer is just passing the exception message and not full exception. Applicable constructor from CopyFileInVmAnswer 
public CopyFileInVmAnswer(CopyFileInVmCommand cmd, Exception e) {
        super(cmd, false, e.getMessage());
    }